### PR TITLE
Changes for SG 2.1 to run with views and GSI

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Enable SG 2.1 to run with views and GSI.
- Set default number of replicas to 0.

### CODE REVIEW

**Comment by raghusarangapani** [comment]
test_overloaded_channel_cache.py needs a fix, its pending https://github.com/couchbase/sync_gateway/issues/3424
Do Not Merge.

File: libraries/testkit/cluster.py [hunk 1]
```diff
@@ -157,6 +169,10 @@ def reset(self, sg_config_path):
             playbook_vars["revs_limit"] = '"revs_limit": {},'.format(revs_limit)
         except KeyError as ex:
             log_info("Keyerror in getting revs_limit{}".format(ex.message))
+
+        # Sleep for a few seconds for the indexes to teardown
+        time.sleep(5)
```
[the remark this answers is not in the document]
Comment (raghusarangapani) [reply libraries/testkit/cluster.py]: The operations seems to be asynchronous. I will keep this in mind and add it if I find a solution.

[the remark this answers is not in the document]
Comment (sridevi-15) [reply libraries/testkit/cluster.py]: sure, you can add TODO : can replace with better solution to wait for index?

**Comment by raghusarangapani** [comment]
I ran 1 test case with this branch,
2.1 - replicas got added to SG config.
2.0 - no replicas/ no views got added.
1.5 - no replicas/ no views got added.

File: libraries/testkit/cluster.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (raghusarangapani) [reply libraries/testkit/cluster.py]: Done

**Pull request merged**